### DISCUSSION TITLE
How to get contained item Url in Razor Summary view

### DISCUSSION THREAD

**Discussion by giantpenguin** [OrchardCMS/OrchardCore#7781] [Q&A] [answered]
This is the snippet to render a BlogPost's link in liquid (Content-BlogPost.Summary.liquid):

`    {% a display_for:Model.ContentItem %}
        <h2 class="post-title">
            {{ Model.ContentItem.DisplayText }}
        </h2>
        <h3 class="post-subtitle">
            {{ Model.ContentItem.Content.BlogPost.Subtitle.Text }}
        </h3>
    {% enda %}
`
How can we do similar in Summary view in Razor?

**Comment by deanmarcussen** [comment, marked as answer]
`<a display-for="@Model.ContentItem">@Model.ContentItem.Content.BlogPost.Subtitle.Text</>`

and a tag helper reference in your `_ViewImports.cshtml` file

`@addTagHelper *, OrchardCore.Contents.TagHelpers`

and a direct reference to `OrchardCore.Contents.TagHelpers` nuget package.

**Comment by giantpenguin** [comment]
Thank @deanmarcussen! 

But it does not render the link, just the text.
I have this in the Summary.cshtml: 
`<h4 class="service-heading"><a display-for="@Model.ContentItem">@Model.ContentItem.DisplayText</a></h4>`

Just see the @Model.ContentItem.DisplayText's text value, no link under it.
What else should I do?

**Comment by kdubious** [comment]
I would think your “a” tag needs an “href” attribute.

**Comment by giantpenguin** [comment]
Thank @kdubious! 
But that is what I want, I do not know what expression to assign to the href attribute to gain the similar result of liquid way.
`<a href="WHAT should be here???">@Model.ContentItem.DisplayText</a>`

**Comment by Skrypt** [comment]
```
{% a display_for: Model %}
{% enda %}
```

**Comment by deanmarcussen** [comment]
> <a display-for="@Model.ContentItem">@Model.ContentItem.Content.BlogPost.Subtitle.Text</>

If this isn't working please share your code.

For infos @kdubious the `display-for` tag helper writes the href attribute

**Comment by giantpenguin** [comment]
Thank you All! Specifically to @deanmarcussen 

It was my false I had @addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers in the _ViewImports
but it must be @addTagHelper *, OrchardCore.Contents.TagHelpers as @deanmarcussen mentioned.
